sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -820,7 +820,7 @@ static size_t _get_l2src(kernel_pid_t iface, uint8_t *l2src, size_t l2src_maxlen
     }
 
     if (try_long && ((res = gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG, 0,
-                                            l2src, l2src_maxlen)) > max_short_len)) {
+                                            l2src, l2src_maxlen)) > (int)max_short_len)) {
```

Comment:
I think without this change this wouldn't collide with #7925. This file is removed in that PR, so if you revert this change it could get in even with the GNRC embargo.